deepspeed/utils/timer.py

```diff
@@ -167,62 +171,71 @@ def __init__(
 
     def update_epoch_count(self):
         self.epoch_count += 1
-        self.local_step_count = 0
+        self.local_microstep_count = 0
 
     def _init_timer(self):
         self.initialized = True
 
     def start(self):
         self._init_timer()
         self.started = True
-        if self.total_step_count >= self.start_step:
+        if self.local_microstep_count >= self.start_step * self.gradient_accumulation_steps:
```

Comment:
Can be `if self.global_step_count >= self.start_step:`

Reply:
`self.start_step` is defined in terms of global steps, so ideally should be compared with that.